src/baseService.ts

```diff
@@ -263,17 +263,17 @@ export class BaseService {
   @nonHTTPRequestMethod
   private _resolveQuery(methodName: string, args: any[]): any {
     const meta = this.__meta__;
-    const query = meta[methodName].query || {};
+    const query = { ...meta[methodName].query };
     const queryParams = meta[methodName].queryParams;
     for (const pos in queryParams) {
-      if (queryParams[pos]) {
+      if (queryParams[pos] && args[pos] !== undefined) {
         query[queryParams[pos]] = args[pos];
       }
     }
     const queryMapIndex = meta[methodName].queryMapIndex;
     if (queryMapIndex >= 0) {
       for (const key in args[queryMapIndex]) {
-        if (args[queryMapIndex][key]) {
+        if (args[queryMapIndex][key] !== undefined) {
```

Comment:
### Issue 3 - discarding legitimate falsy values

While this bit technically was already preventing `undefined` values being added to the config object, what it actually does is filter out all _falsy_ values. Therefore empty strings `""`, number `0`,  bolean `false` etc which can all be legitimate values would be discarded. Using an explicit test for undefined allows legit values to pass through